src/trace_processor/importers/generic_kernel/generic_kernel_parser.cc

```diff
@@ -78,18 +81,42 @@ void GenericKernelParser::ParseGenericTaskStateEvent(
   const uint32_t cpu = static_cast<uint32_t>(task_event.cpu());
   const uint32_t tid = static_cast<uint32_t>(task_event.tid());
   const int32_t prio = task_event.prio();
+  const size_t state = static_cast<size_t>(task_event.state());
 
-  // TODO(jahdiel): Handle the TASK_STATE_CREATED in order to set
-  // the thread's creation timestamp.
-  UniqueTid utid = context_->process_tracker->UpdateThreadName(
-      tid, comm_id, ThreadNamePriority::kGenericKernelTask);
-
-  StringId state_string_id =
-      task_states_[static_cast<size_t>(task_event.state())];
-  if (state_string_id == kNullStringId) {
+  if (state > task_states_.size()) {
     context_->storage->IncrementStats(stats::task_state_invalid);
   }
 
+  StringId state_string_id = task_states_[state];
+
+  // Handle thread creation
+  auto utid_opt = context_->process_tracker->GetThreadOrNull(tid);
```

Comment:
I think you actually want something different here:
1) if you are `created_string_id_`, you should call StartNewThread
2) If you are `destroyed_string_id_` you should call GetThreadOrNull
3) Otherwise you should call GetOrCreateThread.